### PR TITLE
fuzz: remove invalid characters from cluster header names

### PR DESCRIPTION
**Pull request by asraa**
Removes invalid characters from header names in route_fuzz_test
Risk Level: Low
Testing: Corpus entry added
Fixes Issue:
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=15241

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

File: test/common/router/route_fuzz_test.cc [hunk 1]
```diff
@@ -36,13 +36,24 @@ cleanRouteConfig(envoy::api::v2::RouteConfiguration route_config) {
   auto internal_only_headers = clean_config.mutable_internal_only_headers();
   std::for_each(internal_only_headers->begin(), internal_only_headers->end(),
                 [](std::string& n) { n = Fuzz::replaceInvalidCharacters(n); });
-  // Remove invalid characters in each virtual host.
+  // Remove invalid characters in each virtual host and in their routes' cluster header names.
```
Comment (dio) [inline test/common/router/route_fuzz_test.cc]: Can you add a comment which line is exactly doing this?

File: test/common/router/route_fuzz_test.cc [hunk 2]
```diff
@@ -36,13 +36,24 @@ cleanRouteConfig(envoy::api::v2::RouteConfiguration route_config) {
...
+  std::for_each(
+      virtual_hosts->begin(), virtual_hosts->end(),
+      [](envoy::api::v2::route::VirtualHost& virtual_host) {
+        virtual_host = replaceInvalidHeaders<envoy::api::v2::route::VirtualHost>(virtual_host);
+        std::for_each(virtual_host.mutable_routes()->begin(), virtual_host.mutable_routes()->end(),
+                      [](envoy::api::v2::route::Route& route) {
+                        if (route.has_route()) {
+                          route.mutable_route()->set_cluster_header(
+                              Fuzz::replaceInvalidCharacters(route.route().cluster_header()));
+                          if (route.route().cluster_header().empty()) {
```
Comment (dio) [inline test/common/router/route_fuzz_test.cc]: And this one, why are we doing this?

File: test/common/router/route_fuzz_test.cc (hunk 1 above)
Comment (asraa) [reply test/common/router/route_fuzz_test.cc]: Done, thank you!

**Pull request merged**